File: NetworkAccess/app/src/main/java/com/lee/networkaccess/BaseApplication.java
package com.lee.networkaccess;

import android.app.Application;

import jv.lee.library.NetworkManager;

/**
 * @author jv.lee
 * @date 2019/9/23.
 * @description
 */
public class BaseApplication extends Application {
    @Override
    public void onCreate() {
        super.onCreate();
        NetworkManager.getDefault().init(this);
    }
}

File: KotlinLibraryX/library/src/main/java/com/lee/library/base/BaseApplication.kt
package com.lee.library.base

import android.app.Application
import androidx.multidex.MultiDexApplication

/**
 * application通用基类
 * @author jv.lee
 * @date 2020/3/20
 */
abstract class BaseApplication : MultiDexApplication() {

    override fun onCreate() {
        super.onCreate()
        ApplicationExtensions.app = this
        init()
    }

    override fun onTerminate() {
        super.onTerminate()
        unInit()
    }

    protected abstract fun init()
    protected abstract fun unInit()
}

object ApplicationExtensions {
    lateinit var app: Application
}